dashboard/client/components/Sidebar/MenuIcons.tsx

```diff
@@ -0,0 +1,39 @@
+import React from 'react'
+
+import AnalyticsIcon from './AnalyticsIcon'
+import AssignedIcon from './AssignedIcon'
+import MainIcon from './MainIcon'
+import ProfileIcon from './ProfileIcon'
+
+const icons = ['analytics', 'assigned', 'main', 'profile']
+
+const selector = (item, selected) => {
+  switch (item) {
+    case icons[0]:
+      return <AnalyticsIcon selected={selected} />
+    case icons[1]:
+      return <AssignedIcon selected={selected} />
+    case icons[2]:
+      return <MainIcon selected={selected} />
+    case icons[3]:
+      return <ProfileIcon selected={selected} />
+    default:
+      return null
+  }
+}
+
+const MenuIcons = ({ selected, toggleSelected }) => (
+  <>
+    {icons.map(icon => (
+      <div
+        key={icon}
+        onClick={toggleSelected(icon)}
+        className={selected === icon ? 'selected' : ''}
+      >
+        {selector(icon, selected === icon)}
```

Comment:
we can use classnames or clsx here instead of passing the selected prop to all of the svgs:
```
const MenuIcons = ({ selected, toggleSelected }) => (
  <>
    {icons.map(icon => (
      <div
        key={icon}
        onClick={() => toggleSelected(icon)}
        className={clsx({ selected: selected === icon })}
      >
        <Selector item={icon} />
      </div>
    ))}
  </>
);
```
```
.selected {
  fill: #43b02a;
}
```

Reply:
each of the icons (AnalyticsIcon, ProfileIcon, etc.) needs a boolean telling it to switch between two fill colors:

```js
<svg>
  <path
      fill={selected ? '#43B02A' : '#565656'}
  ...
```

any idea how we can do that better?

Reply:
```
const MenuIcons = ({ selected, toggleSelected }) => (
  <>
    {icons.map(icon => (
      <div
        key={icon}
        onClick={toggleSelected(icon)}
        className={clsx({ sideNav: true, selected: selected === icon })}
      >
        <Selector item={icon} />
      </div>
    ))}
  </>
);
```
```
.navIcon {
  fill: #565656
}
.selected {
  fill: #43b02a;
  background-color: #EFEFEE;
}
```
the svg is just inheriting the fill from the div

Reply:
ur so smart 😻 i'm glad you're on my team!